src/naemon/checks_service.c

```diff
@@ -307,6 +307,7 @@ static int run_scheduled_service_check(service *svc, int check_options, double l
 	if (neb_result == NEBERROR_CALLBACKOVERRIDE) {
 		clear_volatile_macros_r(&mac);
 		free_check_result(cr);
+		free(cr);
```

Comment:
shouldn't `free_check_result` handle that?

Reply:
would be ok for me, i just just checked how the other parts handle freeing check_results.

Reply:
Okay, makes sense, thanks for taking a look.

Perhaps we should use nm_free though?

Reply:
you are right, i pushed a new single commit fixing all this

Reply:
Great, looks fine for me once the build passes.